### ISSUE TITLE
Copy PlainText: Use `optional_permissions` for `downloads`

### ISSUE THREAD

**Issue by aaronraimist** [erosman/support#380]
It would be nice if you set the "Download files and read and modify the browser’s download history" permission as an optional permission https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/optional_permissions so users would only have to grant it when they try to export their preferences

**Comment by erosman** [comment]
That is possible. However, downalod permission is such a basic permision.

More Info: https://github.com/erosman/support/issues/375#issuecomment-946374196

**Comment by aaronraimist** [comment]
The ability to download files is fine. It's the "read and modify the browsers download history" part that is slightly concerning to people. I wish WebExtensions separated those but since they don't setting it as an optional permission can make people more comfortable with installing the extension.

As a developer I know I always want to hold the least amount of permissions that I can and still be able to perform the task the user wanted. It is reasonable to be concerned about allowing access to download history as it could contain sensitive information in the filename or URL.

**Comment by erosman** [comment]
Good point.
Let me see what can be done.

**Comment by cmddegi** [comment]
I just wanted to add my +1 to @aaronraimist's post. This extension is the only one i have installed besides uBlock, and i already have it for ages (A big thank you btw!). Since Firefox popped up with the request i asked myself why it would need to see very personal data like download history, and refrained from allowing the update until i have time to look it up.
I guess in light of recurring security breaches sneaking in trough updates (e.g. in Android apps), many people could be concerned and stick with the previous version or even uninstall it.

Thanks again!
Martin

**Comment by erosman** [comment]
I was thinking about the concept of  `optional_permissions`.

New features were added to **Copy PlainText** which required `downloads` permission.
- New features are not wanted ➜ Use v1.12
- New features are wanted ➜ Use v1.13

With `optional_permissions`
- New features are not wanted and `optional_permissions` is not given ➜ Use v1.12
- New features are wanted and `optional_permissions` is given ➜ Use v1.13


In other words, users not giving `optional_permissions` is similar to using v1.12.
Unless users are expecting to pick & choose between add-on features. :thinking:

**Comment by erosman** [comment]
As I am working on the next update, I checked and it seems:

- 90% of users use v1.13
- 9% of user remained on 1.12
- 1% using older versions

**Comment by erosman** [comment]
I have been thinking about this issue.

I am the developer of [FoxyProxy](https://github.com/foxyproxy/browser-extension) which has [optional_permissions](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/optional_permissions) for "privacy" in order to set `privacy.network.webRTCIPHandlingPolicy`.

It makes sense to use "optional_permissions"  there since setting WebRTC is an optional feature and not part of the core functionality of the extension.

However, being able to import & export extension preferences is considered part of the core features of any extension that has user preferences.

**Comment by the-Arioch** [comment]
> However, downalod permission is such a basic permision.

i don't know if that is techincally true, but FF GUI represents it as "download files AND EDIT DOWNLOAD HISTORY" and that sounds a bit different than "download files and log it into download history" which could (or could not) have been meant.

**Comment by the-Arioch** [comment]
Sad...

I wonder if someone created a bug on FF to split it into two...

**Comment by erosman** [comment]
### v1.15
- Changed <code>"&lt;all_urls&gt;"</code> and <code>"downloads"</code> to optional permissions
